src/DurableTask.Netherite/StorageLayer/Faster/LogWorker.cs

```diff
@@ -264,11 +285,17 @@ async IAsyncEnumerable<PartitionUpdateEvent> EventsToReplay(long from)
                         await iter.WaitAsync(this.cancellationToken);
                     }
 
+                    if (this.traceLogDetails)
+                    {
+                        this.TraceLogDetail("Read", iter.NextAddress, new ReadOnlySpan<byte>(result, 0, entryLength));
+                    }
+
                     if ((result[0] & first) != none)
                     {
                         if ((result[0] & last) != none)
                         {
-                            partitionEvent = (PartitionUpdateEvent)Serializer.DeserializeEvent(new ArraySegment<byte>(result, 1, entryLength - 1));
+                            partitionEvent = TryDeserializePartitionEvent(new MemoryStream(result, 1, entryLength - 1), reassembled:false);                 
```

Comment:
I haven't fully grok'ed this codepath, but it appears to me that a single partition event can sometimes be split into several pieces, which are then re-eassmebled using the memoryStream APIs. Is that what this `reassembled` parameter is representing? Essentially, at this point in the code, we haven't fully re-assembled an atomic partition event?

Reply:
yes, see notice above about fragmentation and reassembly. We reassemble all the pieces when we receive the last portion.

Reply:
Got it. So - since we're trying to reconstruct an event from various pieces, if any of them fails to deserialize and there's still more pieces to assemble, shouldn't we abort trying to deserialize the next piece/fragment of this event?

Reply:
We are only trying to deserialize once we got the last portion.

Reply:
Ah, I see that now. Goes on to show that I did not understand this codepath, but I see how it works now. Sounds good, no blocker here then.